pkg/operator/ceph/file/controller.go

```diff
@@ -194,9 +194,15 @@ func (r *ReconcileCephFilesystem) reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, *cephFilesystem, errors.Wrap(err, "failed to add finalizer")
 	}
 
-	// The CR was just created, initializing status fields
+	// The CR was just created, initialize status as 'Progressing'
 	if cephFilesystem.Status == nil {
-		r.updateStatus(k8sutil.ObservedGenerationNotAvailable, request.NamespacedName, k8sutil.EmptyStatus, nil)
+		updatedCephFS := r.updateStatus(k8sutil.ObservedGenerationNotAvailable,
+			request.NamespacedName, cephv1.ConditionProgressing, nil)
+		if updatedCephFS == nil || updatedCephFS.Status == nil {
+			err = errors.Errorf("failed to update ceph filesystem status")
+			return reconcile.Result{}, *cephFilesystem, err
```

Comment:
no need to assign the err on the previous line since this line is returning
```suggestion
			return reconcile.Result{}, *cephFilesystem, errors.Errorf("failed to update ceph filesystem status")
```

Reply:
Done